python/pdo/common/config.py

```diff
@@ -158,18 +156,21 @@ def parse_configuration_file(filename, variable_map) :
 
 # -----------------------------------------------------------------
 # -----------------------------------------------------------------
-def build_configuration_map() :
+def build_configuration_map(**kwargs) :
     """Build a standard representation for the environment configuration.
     The map that is created is generally provided when the configuration
     files are read by parse_configuration_file and parse_configuration_files.
+
+    The following can be overridden through input parameters: 'home', 'ledger',
+    'ledger_type', 'log_level', 'interpreter'.
     """
 
     try :
         # these are the minimum variables that must be defined in the
         # environment; there are reasonable defaults for the others
-        ContractHome = os.environ["PDO_HOME"]
-        LedgerURL = os.environ["PDO_LEDGER_URL"]
-        LedgerType = os.environ["PDO_LEDGER_TYPE"]
+        ContractHome = kwargs.get('home') or os.environ["PDO_HOME"]
```

Comment:
This could be a good opportunity to just use the input args and remove (direct) dependencies to environment variables.

Reply:
agreed though i don't think this is the PR to do it. this brings together a lot of what we have in one place so that it can be adjusted. but removing the dependencies would be a PR you probably wouldn't be happy to review.